backend/lib/input.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+# Process input parameters
+module Input
+
+  # Return text (null if blank)
+  def self.text(str)
+    str = str.to_s.strip
+    str = nil if str == ""
+
+    return str
+  end
+
+  # Return a number (nil or undefined returns 0)
+  def self.number(n)
+    n = n.to_s.to_i
+
+    return n
+  end
+
+  # Return a boolean (1 or nil)
+  def self.boolean(n)
```

Comment:
OK. As mentioned the system is returning <either> "1" <or> nil. I agree "boolean" is a bit misleading.  I can change the name to something else? Maybe "on_off", maybe"one_or_nil"?  Any ideas?

Reply:
UPDATE: This is used process inputs from checkboxes on the sample type form (Note I do not have anything from the front-end for this yet so I just used "1" and "nil" in my test cases).

That said, Mariko and I have talked about checkboxes before and she wants to send "true" in her API calls.  I will keep this as "boolean" and change it to return { true } or { false } - where "true" an "on" will return { true } and anything else to return { false }.  

That is much better. Plus it is consistent with what I used in other checkboxes in other parts of the code.  (I want to allow "on" in addition to "true"  because standard checkboxes in standard HTML forms return "on")

Glad I took a second look at this one. Thanks.

Reply:
"UPDATE:" change done.